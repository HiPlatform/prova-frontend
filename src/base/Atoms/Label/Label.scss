@import "../../../variables";

span {
  font-size: smaller;
  font-weight: bolder;
  color: $darkColor;
}