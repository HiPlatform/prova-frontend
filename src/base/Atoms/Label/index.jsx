import React from 'react';
import PropTypes from 'prop-types';

import './Label.css';

const Label = ({ text }) => (
  <span>{text}</span>
);

Label.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Label;
