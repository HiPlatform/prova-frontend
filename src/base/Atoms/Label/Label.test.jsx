import React from 'react';
import { shallow } from 'enzyme';
import Label from './index';

describe('Label', () => {
  const props = {
    text: 'some label',
  };
  const component = shallow(<Label {...props} />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render the text prop', () => {
    expect(component.text()).toBe(props.text);
  });
});
