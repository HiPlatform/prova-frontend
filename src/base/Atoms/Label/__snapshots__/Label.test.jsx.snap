// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label
    text="some label"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "some label",
    },
    "ref": null,
    "rendered": "some label",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "some label",
      },
      "ref": null,
      "rendered": "some label",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
